### ISSUE TITLE
using ViewModelBuilder through multiple widgets and dialogs.

### ISSUE THREAD

**Issue by casvanluijtelaar** [Stacked-Org/stacked#115]
this is most likely not an issue, I'm probably just understanding things wrong. Say you have an app that basically is just a form. But besides the main dart file where the ViewModelBuilder lives, there are seperate other files containing sub widgets& dialogs to enter more detailed stuff. all these sub widgets and dialogs need to update the state of the main file, as well as yet other sub widgets in their own files. besides that i'm currently using a model to store all that form data. originally I had just an instance of that model in the viewmodel, but that's probably a bad way to do it.

tldr: how to pass data through multiple files and update multiple points in the widget tree accordingly.

**Comment by FilledStacks** [comment]
@casvanluijtelaar you wouldn't want to do that. That will cause a lot of duplicate code that adds no additional value. Create a service that's injected into all the viewmodels that needs to update data and update it in that service. The service will contain all the data, when you're dong you get the data from there or add the functionality for the service to submit that data for you.

**Comment by casvanluijtelaar** [comment]
yeah I can tell I'm doing things wrong. trying to do a half ass implementation. I'd like to structure the app completely around [this](https://github.com/FilledStacks/flutter-tutorials/tree/master/012-provider-architecture-pt2) tutorial. The only thing I'm a bit worried about is instantiating basically all services and models in main(). For larger apps with a lot of different views, services, models... most of them locked behind a login system. Is it a safe and efficient way to approach this? it doesn't seem very scaleable?

**Comment by FilledStacks** [comment]
You definitely shouldn't be instantiating all services in main. they're usually registered as lazy singletons so they're only constructed when they are required the first time.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by casvanluijtelaar** [comment]
right, time for me to read up more on the get_it package specifically. Thanks for the help!
